spark2/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -209,9 +212,15 @@ public String toString() {
     return String.format("IcebergWrite(table=%s, format=%s)", table, format);
   }
 
-  private static class TaskCommit extends TaskResult implements WriterCommitMessage {
-    TaskCommit(TaskResult toCopy) {
-      super(toCopy.files());
+  private static class TaskCommit implements WriterCommitMessage {
+    private final List<DataFile> taskFiles;
```

Comment:
This should be an Array, like it was when this class was based on `TaskResult`. Arrays are easier to handle when working with Serializable classes because we don't have to worry about bugs caused by List implementations (like the recent Kryo bug with unmodifiable lists).

Reply:
Nice finding,  make sense.